src/PluginMiddleware.js

```diff
@@ -0,0 +1,32 @@
+import {sizeof} from './Utility.js'
+
+/**
+ *
+ */
+function run(node, plugins, len) {
+	for (var i = 0; i < len; i++)
+		plugins[i](node)
+}
+
+/**
+ *
+ */
+export function pluginMiddleware(plugins) {
+	return function(ast) {
```

Comment:
because this is a list it prevents us from running a plugin on a whole `ast` right now (without overloading plugins' signature) - maybe worth considering wrapping it in some kind of a root node?

Reply:
Yes, lets have a `root` node.